### PR TITLE
Add draft operator role

### PR DESCRIPTION
**Pull request by alexaunter**
Purpose: more flexible user permissions separation.

The original problem we wanted to solve was to give less privileged users ability to manage draft groups, while being unable to create and terminate lockups, fund draft groups.

The solution is to have two roles in the contract, `operator`(called `deposit` previously) and `draft_operator`

Highlights:
1. introduction of `draft_operator` role. The role can perform all actions on drafts that require authorization: `create_draft_group`, `create_draft`, `discard_draft_group`
2. current `deposit` role is an admin role, as before, it can perform deposit operations `add_lockup`, `fund_draft_group`, as well as all actions `draft_operator` can do, plus user permissions management.
3. any user with `deposit` role can trigger `terminate` on the contract, while user with only `draft_operator` role cannot.
4. `add_to_deposit_whitelist`, `remove_from_deposit_whitelist` method now accepts multiple `account_ids`,

Proposal: since `deposit` role is more of an admin role, it might be better to rename it accordingly, for example to `operator` role.

### CODE REVIEW

File: src/termination.rs [hunk 1]
```diff
@@ -23,19 +23,13 @@ pub struct TerminationConfig {
...
+    ) -> (Balance, AccountId) {
         let termination_config = self
             .termination_config
             .take()
             .expect("No termination config");
-        assert_eq!(
-            termination_config.beneficiary_id.as_ref(),
-            initiator_id,
-            "Unauthorized"
-        );
```
Comment (frol) [inline src/termination.rs:38]: @evgenykuzyakov We decided that any "admin" (a.k.a. "deposit whitelisted user") can terminate lockups, but the funds will be refunded back to the original depositor (be it DAO or some independent account). Is it fine with your use-cases?

**Review by frol — APPROVED** [review]
Looks good to me. There is basically only one controversial change in the behavior I highlighted in an inline comment and if that is fine with @evgenykuzyakov, this is good to me merged in my opinion

File: src/lib.rs [hunk 2]
```diff
@@ -257,22 +278,64 @@ impl Contract {
...
-        self.deposit_whitelist.insert(account_id.as_ref());
+        let account_ids = if let Some(account_ids) = account_ids {
+            account_ids
+        } else {
+            vec![account_id.expect("expected either account_id or account_ids")]
+        };
+        for account_id in account_ids {
+            self.deposit_whitelist.remove(&account_id.into());
+        }
     }
```
Comment (evgenykuzyakov) [inline src/lib.rs:321]: Should we check that `deposit_whitelist` is not empty?

**Review by evgenykuzyakov — APPROVED** [review]

File: src/lib.rs (hunk 2 above)
Comment (frol) [reply src/lib.rs:321]: @alexaunter let’s add this minor safeguard 🙏

Comment (alexaunter) [reply src/lib.rs:321]: Fixed. Should I also disallow initializing contract with empty draft operators list?

**Review by evgenykuzyakov — APPROVED** [review]

**Pull request merged**